coordinator/gscoordinator/cluster_builder.py

```diff
@@ -144,6 +145,8 @@ def __init__(
         self._preemptive = preemptive
         self._vineyard_shared_mem = vineyard_shared_mem
 
+        self._dataset_proxy = dataset_proxy
```

Comment:
I prefer to let the `dataset_proxy` follow the logic of `volumes`, they are both json strings. i.e. 
- loads the str to json in here, not in the `EngineCluster(dataset_proxy=json.loads(dataset_proxy), ...)`; 
- b64encode it in the chart then decode here.